localstack/services/apigateway/invocations.py

```diff
@@ -146,36 +146,45 @@ def authorize_invocation(invocation_context: ApiInvocationContext):
         run_authorizer(invocation_context, authorizer)
 
 
-def validate_api_key(api_key: str, stage: str):
+def validate_api_key(api_key: str, invocation_context: ApiInvocationContext):
 
     usage_plan_ids = []
-
-    client = aws_stack.connect_to_service("apigateway")
+    client = aws_stack.connect_to_service("apigateway", region_name=invocation_context.region_name)
     usage_plans = client.get_usage_plans()
     for item in usage_plans.get("items", []):
         api_stages = item.get("apiStages", [])
-        usage_plan_ids.extend(
-            item.get("id") for api_stage in api_stages if api_stage.get("stage") == stage
-        )
+        for api_stage in api_stages:
+            if (
+                api_stage.get("stage") == invocation_context.stage
+                and api_stage.get("apiId") == invocation_context.api_id
+            ):
+                usage_plan_ids.append(item.get("id"))
 
     for usage_plan_id in usage_plan_ids:
         usage_plan_keys = client.get_usage_plan_keys(usagePlanId=usage_plan_id)
         for key in usage_plan_keys.get("items", []):
             if key.get("value") == api_key:
-                return True
+                # check if the key is enabled
+                api_key = client.get_api_key(apiKey=key.get("id"))
+                return api_key.get("enabled") in ("true", True)
 
     return False
 
 
-def is_api_key_valid(is_api_key_required: bool, headers: Dict[str, str], stage: str):
-    if not is_api_key_required:
+def is_api_key_valid(invocation_context: ApiInvocationContext) -> bool:
+    # https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-api-key-source.html
+    client = aws_stack.connect_to_service("apigateway")
+    rest_api = client.get_rest_api(restApiId=invocation_context.api_id)
+
+    # It means it's the authorizer job to return the API key, and as we're only mocking it, it can't
```

Comment:
can you clarify this comment? Is it related to the line above where we check the api key source or a generic comment about the method?

Reply:
I'll clarify in the PR as well.
Basically, when the `apiKeySource` is set to `AUTHORIZER`, the authorizer is supposed to return the api key as a field (`usageIdentifierKey`): see https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-lambda-authorizer-output.html

Right now I don't have a setup to verify what the actual behaviour is in AWS, but I expected them to validate the key from the authorizer context only after the authorizer has ran if the key returned is valid.

However, as the authorizers are only mocked in community, there is now way to properly validate that, so we always return that the key is valid for simplicity. It's up to us to patch this method in -ext if we want to implement it.